[throwtheswitch-unity] Create a new port

nfarid opened the pull request:
**Describe the pull request**

- #### What does your PR fix?
  Adds the port for [Unity (C unit testing library).](https://github.com/ThrowTheSwitch/Unity)

- #### Which triplets are supported/not supported? Have you updated the [CI baseline](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt)?
  All, I presume.
  No.

- #### Does your PR follow the [maintainer guide](https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/maintainer-guide.md)?
  Yes

- #### If you have added/updated a port: Have you run `./vcpkg x-add-version --all` and committed the result?
  Yes.

nfarid committed:
[throwtheswitch-unity] Create a new port

Adela0814 commented on ports/unity/portfile.cmake:
```diff
@@ -0,0 +1,36 @@
+vcpkg_from_github(
+    OUT_SOURCE_PATH SOURCE_PATH
+    REPO "ThrowTheSwitch/Unity"
+    REF "v2.5.2"
```
```suggestion
    REF "v${VERSION}"
```

Adela0814 commented on ports/unity/portfile.cmake:
```diff
@@ -0,0 +1,36 @@
+
```
Please remove this new line.

Adela0814 commented on ports/unity/portfile.cmake:
```diff
@@ -0,0 +1,36 @@
+vcpkg_cmake_configure(
+    SOURCE_PATH "${SOURCE_PATH}"
+    OPTIONS ${FEATURE_OPTIONS}
+        -DUNITY_EXTENSION_FIXTURE="${UNITY_EXTENSION_FIXTURE}"
+        -DUNITY_EXTENSION_MEMORY="${UNITY_EXTENSION_MEMORY}"
```
```suggestion
        "-DUNITY_EXTENSION_FIXTURE=${UNITY_EXTENSION_FIXTURE}"
        "-DUNITY_EXTENSION_MEMORY=${UNITY_EXTENSION_MEMORY}"
```

Adela0814 commented on ports/unity/usage:
```diff
@@ -0,0 +1,5 @@
+
+The package unity provides CMake targets:
+
+    find_package("unity" REQUIRED)
+    target_link_libraries("main" PRIVATE "unity::framework")
```
The quotation marks here are unnecessary.

Adela0814 commented on ports/unity/usage:
```diff
@@ -0,0 +1,5 @@
+
```
Please remove this line.

Adela0814 commented on ports/unity/vcpkg.json:
```diff
@@ -0,0 +1,26 @@
+{
+  "name": "unity",
```
Please add a prefix for the port name.

Adela0814 commented:
Note: I will be converting your PR to draft status. When you respond, please revert to "ready for review". That way, I can be aware that you've responded since you can't modify the tags.

nfarid commented:
Did the following changes:
- Changed REF to use ${VERSION}
- Moved quotation marks in vcpkg_cmake_configure options
- Remove redundant quotation marks from the usage file
- Changed port's name

Adela0814 commented:
Usage test failed in my side:
failures:
![image](https://github.com/microsoft/vcpkg/assets/95600143/e4fa105e-211a-4752-bd9d-1dcd19815c59)

Adela0814 commented on ports/throwtheswitch-unity/portfile.cmake:
```diff
@@ -0,0 +1,36 @@
+
```
Please remove this blank line.

Adela0814 commented on ports/throwtheswitch-unity/usage:
```diff
@@ -0,0 +1,5 @@
+
```
Please remove this blank line.

Adela0814 commented:
Closing this PR since it seems that no progress is being made. Please ping us to reopen if work is still being done.
